tests/wicked/before_test.pm

```diff
@@ -88,6 +88,13 @@ sub run {
             }
             assert_script_run('./autogen.sh ',       timeout => 600);
             assert_script_run('make ; make install', timeout => 600);
+        } elsif (my $wicked_repo = get_var('WICKED_REPO')) {
+            zypper_ar($wicked_repo, name => 'wicked_repo', no_gpg_check => 1);
+            zypper_call('zypper in --force -y --repo wicked_repo --from wicked_repo --allow-vendor-change  --allow-downgrade  wicked', log => 1);
```

Comment:
I had an idea couple of days ago (Rather years, but only did something about this week), 

https://github.com/foursixnine/os-autoinst-distri-opensuse/blob/oopsitsbrokenagain/tests/foursixnine/install_and_reboot.pm#L34. With your permission, I'll steal some from your impl xD
![](https://media.giphy.com/media/t8dgAcwIMa5dZzNhqU/giphy.gif)

Reply:
sure go ahead. Maybe we should just put this in a utility function and call that from both sides?